pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -787,3 +717,42 @@ def longest_common_subsequence(seq1: OneDimensionalArray, seq2: OneDimensionalAr
                     check_mat[i][j] = check_mat[i][j-1]
 
     return OneDimensionalArray(seq1._dtype, check_mat[row][col][-1])
+
+def longest_increasing_subsequence(seq: OneDimensionalArray) -> int:
+    """
+    Finds the length of longest increasing subsequence of the given sequences.
+    Parameters
+    ========
+    seq: OneDimensionalArray
+        The sequence.
+    Returns
+    =======
+    output: int
+        The length of longest increasing subsequence.
+    Examples
+    ========
+    >>> from pydatastructs import longest_increasing_subsequence as LIS, OneDimensionalArray as ODA
+    >>> arr1 = [1, 3, 9, 4, 6]
+    >>> lis = LIS(arr1)
+    >>> str(lis)
+    '4'
+    >>> arr1 = [8, 7, 2, 12, 16, 28, 44]
+    >>> lis = LIS(arr1)
+    >>> str(lis)
+    '5'
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/Longest_increasing_subsequence
+
+    """
+    n=len(seq)
+    lis=[1]*n
+    for i in range(1, n):
+        for j in range(0, i):
+            if seq[i] > seq[j] and lis[i]< lis[j] + 1:
+                lis[i] = lis[j]+1
```

Comment:
It is still O(N^2). Please implement O(N log N) version of this algorithm. Update in this PR only. Do not make new PRs.

Reply:
Sure